src/devices/Card/CreditCard/CreditCard.cs

```diff
@@ -0,0 +1,697 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Common;
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+
+namespace Iot.Device.Card.CreditCardProcessing
+{
+    /// <summary>
+    /// The Credit Card class
+    /// </summary>
+    public class CreditCard
+    {
+        // APDU commands are using 5 elements in this order
+        // Then the data are added
+        private const byte Cla = 0x00;
+        private const byte Ins = 0x01;
+        private const byte P1 = 0x02;
+        private const byte P2 = 0x03;
+        private const byte Lc = 0x04;
+
+        private const int MaxBufferSize = 260;
+
+        private CardWriteRead _nfc;
+        private bool _alreadyReadSfi = false;
+        private byte _target;
+
+        // This is a string "1PAY.SYS.DDF01" (PPSE) to select the root directory
+        private readonly byte[] RootDirectory1 = { 0x31, 0x50, 0x41, 0x59, 0x2e, 0x53, 0x59, 0x53, 0x2e, 0x44, 0x44, 0x46, 0x30, 0x31 };
+        // This is a string "2PAY.SYS.DDF01" (PPSE) to select the root directory
+        private readonly byte[] RootDirectory2 = { 0x32, 0x50, 0x41, 0x59, 0x2e, 0x53, 0x59, 0x53, 0x2e, 0x44, 0x44, 0x46, 0x30, 0x31 };
+
+        /// <summary>
+        /// The size of the tailer elements. Some readers add an extra byte
+        /// usually 0x00 especially NFC ones. While Smart Card readers usually do not
+        /// </summary>
+        public int TailerSize { get; set; }
+
+        /// <summary>
+        /// A list of Tags that is contained by the Credit Card
+        /// </summary>
+        public List<Tag> Tags { get; internal set; }
+
+        /// <summary>
+        /// The list of log entries in binary format
+        /// </summary>
+        public List<byte[]> LogEntries { get; internal set; }
+
+        /// <summary>
+        /// Create a Credit Card class
+        /// </summary>
+        /// <param name="nfc">A compatible Card reader</param>
+        /// <param name="target">The target number as some readers needs it</param>
```

Comment:
please add a remark where do user take this from (unless it's required to read a spec to use this class then fine)